src/boot/chid.h

```diff
@@ -45,20 +55,40 @@ assert_cc(offsetof(Device, descriptor) == 0);
 assert_cc(offsetof(Device, chid) == 4);
 assert_cc(offsetof(Device, devicetree.name_offset) == 20);
 assert_cc(offsetof(Device, devicetree.compatible_offset) == 24);
+assert_cc(offsetof(Device, fw.name_offset) == 20);
+assert_cc(offsetof(Device, fw.compatible_offset) == 24);
 assert_cc(sizeof(Device) == 28);
 
 static inline const char* device_get_name(const void *base, const Device *device) {
-        if (device->descriptor != DEVICE_DESCRIPTOR_DEVICETREE)
+        uint32_t off = 0;
```

Comment:
i'd use `size_t` for this, which is the type we internally use for all memory offsets, according to the C spec. yes, in the struct this is a 32bit entity only, but I think once we read it from the struct we should convert it to our native width.

Reply:
Addressed this as part of https://github.com/systemd/systemd/pull/35747 .